core/src/main/java/org/apache/carbondata/core/view/MVManager.java

```diff
@@ -181,32 +181,51 @@ public void deleteSchema(String databaseName, String viewName) throws IOExceptio
    */
   public MVCatalog<?> getCatalog(
       MVCatalogFactory<?> catalogFactory,
-      boolean reload) throws IOException {
+      List<MVSchema> currSchemas) throws IOException {
     MVCatalog<?> catalog = this.catalog;
-    if (reload || catalog == null) {
-      synchronized (lock) {
-        catalog = this.catalog;
-        if (reload || catalog == null) {
-          catalog = catalogFactory.newCatalog();
-          List<MVSchema> schemas = getSchemas();
-          if (null == catalog) {
-            throw new RuntimeException("Internal Error.");
+    synchronized (lock) {
+      catalog = this.catalog;
+      if (catalog == null) {
+        catalog = catalogFactory.newCatalog();
+      }
+      List<MVSchema> schemas = getSchemas();
+      if (schemas.size() == currSchemas.size() && currSchemas.containsAll(schemas)) {
+        return catalog;
+      }
+      if (null == catalog) {
```

Comment:
check for null catalog before returning catalog. Already a null check has happened right, is there any way that catalog becomes null parallely?

Reply:
as it's inside the sync block, no need to check again for null. removed the check.